libr/util/log.c

```diff
@@ -31,3 +29,24 @@ R_API void r_log_error(const char *str) {
 R_API void r_log_progress(const char *str, int percent) {
 	printf ("%d%%: %s\n", percent, str);
 }
+
+R_API void r_vlogf(RLogLevel level, const char *fmt, va_list ap) {
+	static const char *headers[R_LOG_MAX_VALUE] = {
+		[R_LOG_DEBUG] = "DEBUG: ",
+		[R_LOG_INFO] = "INFO: ",
+		[R_LOG_WARNING] = "WARNING: ",
+		[R_LOG_ERROR] = "ERROR: ",
+		[R_LOG_CRITICAL] = "CRITICAL: ",
+	};
+
+	const char *hdr = R_BETWEEN (0, level, R_LOG_MAX_VALUE - 1) ? headers[level] : "";
+	eprintf (hdr);
```

Comment:
if you refuse to use format strings properly, at least switch to `fputs (hdr, stderr);` so it is consistent with the vfprintf below

Reply:
switched to `eprintf("%s", hdr)`, which is a bit like doing `fprintf(stderr, "%s", "Hello World");` instead of `fprintf(stderr, "Hello World");`